consensus/state.go

```diff
@@ -1033,31 +1074,69 @@ func (cs *ConsensusState) AddProposalBlockPart(height int, part *types.Part) (ad
 		log.Info("Received complete proposal", "hash", cs.ProposalBlock.Hash())
 		if cs.Step == RoundStepPropose && cs.isProposalComplete() {
 			// Move onto the next step
-			go cs.EnterPrevote(height, cs.Round)
+			go cs.EnterPrevote(height, cs.Round, false)
 		} else if cs.Step == RoundStepCommit {
 			// If we're waiting on the proposal block...
-			cs.tryFinalizeCommit(height)
+			cs.tryFinalizeCommit(height, cs.Round)
```

Comment:
+2/3 precommits in addVote() -> EnterCommit(height, commitRound (e.g. vote.Round)), so then we're in commit, but we may not have the whole committed block, so we still block on AddProposalBlockPart()...  vote.Round wasn't necessarily cs.Round.